srtcore/tsbpd_time.cpp

```diff
@@ -218,27 +221,33 @@ CTsbpdTime::time_point CTsbpdTime::getTsbPdTimeBase(uint32_t timestamp_us) const
     return (m_tsTsbPdTimeBase + microseconds_from(carryover_us));
 }
 
-CTsbpdTime::time_point CTsbpdTime::getPktTsbPdTime(uint32_t usPktTimestamp) const
+CTsbpdTime::time_point CTsbpdTime::getTsbPdTimeBase(uint32_t timestamp_us) const
 {
-    time_point value = getPktTsbPdBaseTime(usPktTimestamp) + m_tdTsbPdDelay + microseconds_from(m_DriftTracer.drift());
+    SharedLock lck(m_mtxRW);
+    return getTsbPdTimeBaseNoLock(timestamp_us);
+}
 
-    /*
-    HLOGC(brlog.Debug, log << "getPktTsbPdTime:"
-            << " BASE=" << FormatTime(m_tsTsbPdTimeBase)
-            << " TS=" << usPktTimestamp << "us, lat=" << FormatDuration<DUNIT_US>(m_tdTsbPdDelay)
-            << " DRF=" << m_DriftTracer.drift() << "us = " << FormatTime(value));
-            */
+CTsbpdTime::time_point CTsbpdTime::getPktTsbPdTime(uint32_t usPktTimestamp) const
+{
+    SharedLock lck(m_mtxRW);
+    time_point value = getPktTsbPdBaseTimeNoLock(usPktTimestamp) + m_tdTsbPdDelay + microseconds_from(m_DriftTracer.drift());
 
     return value;
 }
 
+CTsbpdTime::time_point CTsbpdTime::getPktTsbPdBaseTimeNoLock(uint32_t usPktTimestamp) const
```

Comment:
What's the use of this function except for being a part of the expression?

Reply:
A proper calculation of a packet's base TSBPD time (without SRT buffering delay, aka latency).

Reply:
Ok, I can see it's used in two places, but then "Base" in the name here is kinda superfluous, and misleading.

Reply:
Some better term for this value has to be developed. We have:
* Time Base: the time when the connection has started and since when all the time is calculated. Doesn't change during transmission
* Local Time Base - or Shifted Time Base, Earliest Time: The "timespace?" for the time given in 32-bit timestamps which is a sum of the Time Base and N values of the "segment" (which represent a maximum 32-bit time value).

Might be also something like "timespace" or "rollover time base". "Base Time" is simply too similar to "Time Base".

Reply:
Moving this discussion to #3043 to unblock the fix itself.